src/DynamoCore/Core/CustomNodeManager.cs

```diff
@@ -276,13 +276,29 @@ protected virtual void OnCustomNodeRemoved(Guid functionId)
                 if (!disposed)
                     InfoUpdated -= infoUpdatedHandler;
             };
+
+        }
+
+        private static bool ValidateInputSymbolNames(Function node, CustomNodeWorkspaceModel workspace)
+        {
+            foreach (var s in workspace.Nodes.OfType<Symbol>())
+            {
+                if (!s.Parameter.NameIsValid)
+                {
+                    node.Warning("This custom node contains an input with an invalid input symbol.\n" +
+                                 "Please fix the input symbol before saving the custom node.");
```

Comment:
Since this message will be displayed to the user above node, would you put it into resource dictionary. You can find them by right click on **DynamoCore** Project-> **Properties** -> **Resources**
![image](https://user-images.githubusercontent.com/3942418/51012504-6ee83700-152b-11e9-81a5-b9c0dc34bd56.png)

You can always give it a name and added it to the bottom of the list, once added the list will be sorted for you, and make sure you build the solution again. Then you can reference it in the code using `Resources.UnknownVersion`.

Reply:
@QilongTang Yes, I will before I remove the [WIP] tag.